source/stylist/configuration.py

```diff
@@ -61,17 +76,30 @@ def language_lookup(cls, key: str) -> Type[SourceTree]:
 
     @classmethod
     def preprocessor_tags(cls):
+        """
+        Gets the preprocessors understood by this configuration.
+        """
         return cls._PREPROCESSOR_MAP.keys()
 
     @classmethod
     def preprocessor_lookup(cls, key: str) -> Type[TextProcessor]:
+        """
+        Gets the class associated with a particular preprocessor.
+
+        :param key: Preprocessor identifier as returned by `preprocessor_tags`.
+        """
         return cls._PREPROCESSOR_MAP[key]
 
     @classmethod
     def parse_pipe_description(cls, string: str) \
         -> Tuple[str,
                  Type[SourceTree],
                  Sequence[Type[TextProcessor]]]:
+        """
+        Converts a pipline description into classes.
+
+        :param string: Pipeline description, colon separated.
```

Comment:
I would suggest "colon-separated" as it refers to "Pipeline description".

Reply:
Re-ordered as well.